openstack/02a-MAClow/IEEE802154E.c

```diff
@@ -550,11 +551,24 @@ port_INLINE void activity_synchronize_endOfFrame(PORT_RADIOTIMER_WIDTH capturedT
       ieee154e_vars.dataReceived->l2_frameType = ieee802514_header.frameType;
       ieee154e_vars.dataReceived->l2_dsn       = ieee802514_header.dsn;
       memcpy(&(ieee154e_vars.dataReceived->l2_nextORpreviousHop),&(ieee802514_header.src),sizeof(open_addr_t));
-      
-      // toss the IEEE802.15.4 header -- this does not include IEs as they are processed 
+
+      // This is the case where we first hear an EB and do not yet know the ASN used

```

Comment:
Yes, this is good to separate network with different keys. Note that the comment relates to the implicit ASN. EB contains the explicit ASN which we can use but it still does not provide us with any guarantee against replay attack. The problem here is that you should parse the EB to get ASN before you have it passed the security check, which normally one shouldn't do. I will see to add a join-specific call to get this done, as discussed on the 6tisch list...